treewide: Remove nRF Cloud Alerts from NCS (#28276)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

Libraries for networking
------------------------

* Removed nRF Cloud Alerts.

>   As part of the migration to nRF Cloud powered by Memfault, migrating nRF Cloud's Alerts would be redundant since Memfault's Trace Events feature is an option that covers what the alerts previously achieved plus debugging capabilities.

Reviewer: ```suggestion
  As part of the migration to *nRF Cloud powered by Memfault*, the nRF Cloud Alerts feature is now redundant.
  Memfault's Trace Events feature replaces the Alerts feature, as it provides equivalent functionality for event reporting and it also adds enhanced debugging capabilities that were not available with Alerts.
```

Libraries for networking
------------------------

> * Removed nRF Cloud Alerts.

Reviewer 2: ```suggestion
* Removed the nRF Cloud Alerts library.
```

  As part of the migration to *nRF Cloud powered by Memfault*, the nRF Cloud Alerts feature is now redundant.

>   Memfault's Trace Events feature replaces the Alerts feature, as it provides equivalent functionality for event reporting and it also adds enhanced debugging capabilities that were not available with Alerts.

Reviewer 3: ```suggestion
  Memfault's Trace Events feature replaces the Alerts feature, as it provides equivalent functionality for event reporting, and it also adds enhanced debugging capabilities that were not available with Alerts.
```

Libraries for networking
------------------------

> * Removed nRF Cloud Alerts. Since Memfault was introduced as the primary observability layer for nRF Cloud, Alerts is now redundant.

Reviewer: ```suggestion
* Removed nRF Cloud Alerts.
  Since Memfault was introduced as the primary observability layer for nRF Cloud, Alerts is now redundant.
```

Author: @FrancescoSer I decided to add more context in here, could you double check it?
